satpy/composites/__init__.py

```diff
@@ -283,6 +284,58 @@ def apply_modifier_info(self, origin, destination):
                 elif o.get(k) is not None:
                     d[k] = o[k]
 
+    def remove_coords(self, data_arrays, coords=('x', 'y', 'time')):
+        new_data_arrays = []
+        for coord in coords:
+            for ds in data_arrays:
+                ds = ds.copy()
+                if coord in ds.coords:
+                    del ds.coords[coord]
+                new_data_arrays.append(ds)
+        return new_data_arrays
+
+    def check_areas(self, data_arrays, adjust_coords=True):
+        if len(data_arrays) == 1:
+            return data_arrays
+
+        if not all(x.shape == data_arrays[0].shape for x in data_arrays[1:]):
```

Comment:
shouldn't we be checking only the `x` and `y` dimension sizes ? and what guarantees that all data array have the same number of dimensions ?

Reply:
Good point. This was done assuming we were dealing with 2D reader datasets. Maybe I'm not completely xarray-minded yet.